integration_tests/src/main/python/cache_test.py

```diff
@@ -28,3 +32,62 @@ def test_passing_gpuExpr_as_Expr():
             .cache()
             .limit(50)
     )
+
+conf={"spark.rapids.sql.explain":"ALL"}
+@pytest.mark.xfail(reason="TODO: github issue")
+@pytest.mark.parametrize('data_gen', all_gen_no_nulls, ids=idfn)
```

Comment:
I wanted to avoid nulls in a join because I remember some conversation in our standup regarding joining on nulls not working properly. I will add nulls once I have sorted out the other issues.